android/src/main/java/com/rnlib/adyen/AdyenPaymentModule.kt

```diff
@@ -230,6 +230,16 @@ class AdyenPaymentModule(private var reactContext : ReactApplicationContext) : R
         })
     }
 
+    /***
+     * Some user payment methods are using PaymentMethodTypes.GOOGLE_PAY_LEGACY as component type
+     * but from Adyen 4.0.0 we are using PaymentMethodTypes.GOOGLE_PAY, for the old users the
+     * component won't match, so we are comparing the user payment method with both types.
+     */
+    private fun isGooglePlayComponentValid(userPaymentMethod: String?, component: String): Boolean {
```

Comment:
Seems like `userPaymentMethod` does not need to be nullable.
Can be expressed as 

```
private fun isGooglePlayComponentValid(userPaymentMethod: String, component: String): Boolean {
    ...
}
```

Reply:
paymentMethod.type could be null, but I can apply the same as before:

val paymentMethodType = paymentMethod.type ?: ""